Marlin/Marlin_main.cpp

```diff
@@ -4734,9 +4746,15 @@ inline void gcode_M42() {
         safe_delay(500);
       }
     }
-    else // single pins report
+        
+    if ( !(code_seen('P') || code_seen('W') || code_seen('E')))   // single pins report
       for (int8_t pin = first_pin; pin <= last_pin; pin++)
-        report_pin_state(pin);
+          report_pin_state(pin);
+    
+    if (code_seen('E')) {
+      endstop_monitor_flag = endstop_monitor_flag ? false : true;
+      SERIAL_PROTOCOLLN((endstop_monitor_flag ? "endstop monitor active" : "endstop monitor disabled"));
```

Comment:
Size is more important than performance here.
To optimize string storage / program size, change to:
```cpp
SERIAL_PROTOCOLPGM("endstop monitor ");
SERIAL_PROTOCOL(endstop_monitor_flag ? "en" : "dis");
SERIAL_PROTOCOLLNPGM("abled");
```